infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/fsm/InLogEntrySyncState.java

```diff
@@ -74,6 +74,15 @@ public LogReplicationState processEvent(LogReplicationEvent event) throws Illega
             case REPLICATION_SHUTDOWN:
                 cancelLogEntrySync("replication terminated.");
                 return fsm.getStates().get(LogReplicationStateType.STOPPED);
+            case LOG_ENTRY_SYNC_REPLICATED:
+                // Verify the replicated entry corresponds to the current log entry sync cycle (and not a previous/old one)
+                if (transitionEventId.equals(event.getMetadata().getRequestId())) {
+                    log.debug("Log Entry Sync ACK, update last ack timestamp to {}", event.getMetadata().getLastLogEntrySyncedTimestamp());
+                    fsm.setAckedTimestamp(event.getMetadata().getLastLogEntrySyncedTimestamp());
```

Comment:
I don't think they can come out of order, because the receiver can only process in order and it will send a summarized ack, plus the receiver is single threaded in its processing.

Reply:
Yeah, I see the previous logic was doing the same thing. I was thinking about things like network delay

Reply:
Can you explain why we need this check?
Also, now the eventID is a randomUUID, we should replace that with a deterministic method and guarantee no duplicates.

Reply:
Like in the comment, it is needed to:

 // Verify the replicated entry corresponds to the current log entry sync cycle (and not a previous/old one)

The first time we enter LOG_ENTRY_SYNC it is random (because we need an initial one) after that ACKs are always stamped with the same one. It allows to discard an event originated by an old replication cycle, which is not the current one.